jetty-servlet/src/test/java/org/eclipse/jetty/servlet/StatisticsServletTest.java

```diff
@@ -0,0 +1,217 @@
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.DefaultHandler;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringReader;
+import java.nio.ByteBuffer;
+
+/**
+ *
+ */
+public class StatisticsServletTest
+{
+    private Server _server;
+
+    private LocalConnector _connector;
+
+    @Before
+    public void createServer()
+    {
+        _server = new Server();
+
+        _connector = new LocalConnector( _server );
+        _server.addConnector( _connector );
+
+    }
+
+
+    @After
+    public void destroyServer()
+        throws Exception
+    {
+        _server.stop();
+        _server.join();
+    }
+
+    @Test
+    public void get_stats()
+        throws Exception
+    {
+        ContextHandlerCollection contexts = new ContextHandlerCollection();
+        _server.setHandler( contexts );
+
+        StatisticsHandler statsHandler = new StatisticsHandler();
+
+        ServletContextHandler statsContext = new ServletContextHandler( _server, "/" );
+        statsContext.setHandler( statsHandler );
+
+        _server.setHandler( statsContext );
+
+        statsContext.addServlet( new ServletHolder( new TestServlet() ), "/test1" );
+
+        ServletHolder servletHolder = new ServletHolder( new StatisticsServlet() );
+        servletHolder.setInitParameter( "restrictToLocalhost", "false" );
+
+        statsContext.addServlet( servletHolder, "/stats" );
+
+        statsContext.setSessionHandler( new SessionHandler() );
+
+        _server.start();
+
+        String response = getResponse("/test1" );
+
+        response = getResponse("/stats?xml=true" );
+
+        Stats stats = parseStats( response );
+
+        Assert.assertEquals(1, stats.responses2xx);
+
+        response = getResponse("/stats?statsReset=true" );
+
+        response = getResponse("/stats?xml=true" );
+
+        stats = parseStats( response );
+
+        Assert.assertEquals(1, stats.responses2xx);
+
+        getResponse("/test1" );
+
+        getResponse("/nothing" );
+
+        response = getResponse("/stats?xml=true" );
+
+        stats = parseStats( response );
+
+        Assert.assertEquals(3, stats.responses2xx);
+
+        Assert.assertEquals(1, stats.responses4xx);
+
+    }
+
+    public String getResponse( String path )
+        throws Exception
+    {
+        HttpTester.Request request = new HttpTester.Request();
+        request.setMethod( "GET" );
+        request.setURI( path );
+        request.setVersion( HttpVersion.HTTP_1_1 );
+        request.setHeader( "Host", "test" );
+
+        ByteBuffer responseBuffer = _connector.getResponse( request.generate() );
+        return HttpTester.parseResponse( responseBuffer ).getContent();
+    }
+
+
+    public Stats parseStats( String xml )
+        throws Exception
```

Comment:
I think you can implement this method with 2 lines of XPath, using only JDK classes.